provisioning_templates/iPXE/kickstart_default_ipxe.erb

```diff
@@ -16,21 +16,38 @@ oses:
   <% static = (@host.token.nil? ? '?' : '&') + static_arg -%>
 <% end -%>
 
+<%
+options = []
+hostname = @host.name
+iface = @host.provision_interface
+mac = @host.provision_interface.mac
+-%>
+
 <% stage2 = host_param('kickstart_liveimg') ? 'inst.stage2=' + @host.operatingsystem.medium_uri(@host).to_s : '' %>
 
-<%- if (@host.operatingsystem.name.match(/Fedora/i) && @host.operatingsystem.major.to_i < 17) || @host.operatingsystem.major.to_i < 7 %>
-<%- net_options = 'ip=${netX/ip} netmask=${netX/netmask} gateway=${netX/gateway} dns=${dns}' -%>
-<%- else -%>
-<%- net_options = 'ip=${netX/ip}::${netX/gateway}:${netX/netmask}:::none nameserver=${dns}' -%>
-<%- end -%>
+<%
+  if (@host.operatingsystem.name.match(/Fedora/i) && @host.operatingsystem.major.to_i < 17) || @host.operatingsystem.major.to_i < 7
+    options.push('ip=${netX/ip} netmask=${netX/netmask} gateway=${netX/gateway} dns=${dns}')
+    if iface.tag.present? && iface.attached_to.present?
+      options.push("vlanid=#{iface.tag} ksdevice=#{iface.identifier}")
+    end
+  else
+    options.push("ip=${netX/ip}::${netX/gateway}:${netX/netmask}:#{hostname}:#{interface}:none nameserver=${dns}")
```

Comment:
Here is the second error: `interface` is not known variable. You mean `iface` instead.